packages/siopv2-oid4vp-rp-auth/src/functions.ts

```diff
@@ -118,39 +127,71 @@ export async function createRPBuilder(args: {
     )
-
     .withEventEmitter(eventEmitter)
     .withSessionManager(rpOpts.sessionManager ?? new InMemoryRPSessionManager(eventEmitter))
     .withClientMetadata(rpOpts.clientMetadataOpts ?? defaultClientMetadata, PropertyTarget.REQUEST_OBJECT)
-
-    .withCheckLinkedDomain(didOpts.checkLinkedDomains ?? CheckLinkedDomain.IF_PRESENT)
+    .withVerifyJwtCallback(
+      rpOpts.verifyJwtCallback
+        ? rpOpts.verifyJwtCallback
+        : getVerifyJwtCallback(resolver, {
+            wellknownDIDVerifyCallback: getWellKnownDIDVerifyCallback(rpOpts.didOpts, context),
+            checkLinkedDomain: 'if_present',
+          }),
+    )
     .withRevocationVerification(RevocationVerification.NEVER)
     .withPresentationVerification(getPresentationVerificationCallback(didOpts.idOpts, context))
-
-  if (!rpOpts.clientMetadataOpts?.subjectTypesSupported) {
+  if (hasher) {
+    builder.withHasher(hasher)
+  }
+  //fixme: this has been removed in the new version of did-auth-siop
+  /*if (!rpOpts.clientMetadataOpts?.subjectTypesSupported) {
     // Do not update in case it is already provided via client metadata opts
     didMethods.forEach((method) => builder.addDidMethod(method))
-  }
-  builder.withWellknownDIDVerifyCallback(getWellKnownDIDVerifyCallback(didOpts, context))
+  }*/
+  //fixme: this has been removed in the new version of did-auth-siop
+  // builder.withWellknownDIDVerifyCallback(getWellKnownDIDVerifyCallback(didOpts, context))
 
   if (definition) {
     builder.withPresentationDefinition({ definition }, PropertyTarget.REQUEST_OBJECT)
   }
 
-  const key = resolution.key
+  //const key = resolution.key
+  //fixme: this has been removed in the new version of did-auth-siop
+  //builder.withSuppliedSignature(SuppliedSigner(key, context, getSigningAlgo(key.type) as unknown as KeyAlgo), did, kid, getSigningAlgo(key.type))
 
-  if (isManagedIdentifierDidResult(resolution)) {
+  /*if (isManagedIdentifierDidResult(resolution)) {
     //fixme: only accepts dids in version used. New SIOP lib also accepts other types
     builder.withSuppliedSignature(
       SuppliedSigner(key, context, getSigningAlgo(key.type) as unknown as KeyAlgo),
       resolution.did,
       resolution.kid,
       getSigningAlgo(key.type),
     )
-  }
+  }*/
   return builder
 }
 
+function getVerifyJwtCallback(
+  resolver?: Resolvable,
+  verifyOpts?: JWTVerifyOptions & {
+    checkLinkedDomain: 'never' | 'if_present' | 'always'
+    wellknownDIDVerifyCallback?: VerifyCallback
+  },
+): VerifyJwtCallback {
+  return async (jwtVerifier, jwt) => {
+    resolver = resolver ?? getResolver({ subjectSyntaxTypesSupported: ['ethr', 'ion'] })
+    const audience =
+      jwtVerifier.type === 'request-object'
+        ? (verifyOpts?.audience ?? getAudience(jwt.raw))
+        : jwtVerifier.type === 'id-token'
+          ? (verifyOpts?.audience ?? getAudience(jwt.raw))
+          : undefined
+
+    await verifyDidJWT(jwt.raw, resolver, { audience, ...verifyOpts })
+    return true
```

Comment:
Please add a fixme if you are creating code like this. If this gets overlooked we are simply not actually validating anymore

Reply:
added a todo for both places that we're calling the verify function

Reply:
What is up with the code being reverted to the old state again, depending on the verifyDidJwt instead of the signature service :(

Reply:
discussed this, for now returning `true` and added a todo with the ticket number for refactoring it

Reply:
Haven't looked. But not validating is a FIXME of course not a TODO

Reply:
Okay apparently it is marked as fixme. Please be explicit when talking about something; as this only provokes unnecessary discussion and work

Reply:
I've misspoken. it is in fact a fixme